lib/mix/test/mix/tasks/compile_test.exs

```diff
@@ -34,6 +34,25 @@ defmodule Mix.Tasks.CompileTest do
     assert Mix.Tasks.Compile.manifests() |> Enum.map(&Path.basename/1) == [".compile.elixir"]
   end
 
+  test "compile a project with files containing only comments or newlines" do
```

Comment:
Please see "does not recompile empty files" in compile.elixir_test.exs. I think we can write the same test, just add some comments.

Reply:
Ah cool, I'll use that test instead.